File: r-app/src-tauri/src/models/endpoint.rs
use serde::{Deserialize, Serialize};

/// 端点（上游 API 提供方）。对应 `endpoints` 表。
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Endpoint {
    pub id: i64,
    pub name: String,
    pub api_url: String,
    pub api_key: String,
    /// 认证模式：api_key（默认）/ auth_token 等。
    pub auth_mode: String,
    pub enabled: bool,
    /// 是否经全局代理出网（代理地址见 AppConfig.proxy_url）。
    pub use_proxy: bool,
    /// 转换器名：claude / openai。
    pub transformer: String,
    /// 可选锁定模型：非空则强制覆盖客户端请求的 model（专用型端点）；空则透传。
    pub model: String,
    /// 对外暴露/已选的模型清单（聚合型端点，供 /v1/models 公布与 UI 展示）。
    pub models: Vec<String>,
    pub remark: String,
    pub sort_order: i64,
    /// 测试状态：unknown / available / unavailable。
    pub test_status: String,
    pub created_at: String,
    pub updated_at: String,
}

/// 端点多凭证（Token 轮换的凭证项）。对应 `endpoint_credentials` 表。
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct EndpointCredential {
    pub id: i64,
    pub endpoint_id: i64,
    pub api_key: String,
    pub enabled: bool,
    pub sort_order: i64,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateEndpointRequest {
    pub name: String,
    pub api_url: String,
    #[serde(default)]
    pub api_key: String,
    #[serde(default = "default_auth_mode")]
    pub auth_mode: String,
    #[serde(default = "default_true")]
    pub enabled: bool,
    #[serde(default)]
    pub use_proxy: bool,
    #[serde(default = "default_transformer")]
    pub transformer: String,
    #[serde(default)]
    pub model: String,
    #[serde(default)]
    pub models: Vec<String>,
    #[serde(default)]
    pub remark: String,
}

#[derive(Debug, Clone, Default, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateEndpointRequest {
    pub name: Option<String>,
    pub api_url: Option<String>,
    pub api_key: Option<String>,
    pub auth_mode: Option<String>,
    pub enabled: Option<bool>,
    pub use_proxy: Option<bool>,
    pub transformer: Option<String>,
    pub model: Option<String>,
    pub models: Option<Vec<String>>,
    pub remark: Option<String>,
}

fn default_true() -> bool {
    true
}
fn default_auth_mode() -> String {
    "api_key".to_string()
}
fn default_transformer() -> String {
    "claude".to_string()
}

File: r-app/src/services/modules/endpoint.ts
import { request } from "../request";

export interface Endpoint {
  id: number;
  name: string;
  apiUrl: string;
  apiKey: string;
  authMode: string;
  enabled: boolean;
  useProxy: boolean;
  transformer: string;
  model: string;
  models: string[];
  remark: string;
  sortOrder: number;
  testStatus: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEndpointRequest {
  name: string;
  apiUrl: string;
  apiKey?: string;
  authMode?: string;
  enabled?: boolean;
  useProxy?: boolean;
  transformer?: string;
  model?: string;
  models?: string[];
  remark?: string;
}

export type UpdateEndpointRequest = Partial<CreateEndpointRequest>;

export interface EndpointTestResult {
  success: boolean;
  status: string;
  latencyMs: number;
  message: string;
}

export const endpointApi = {
  list: () => request<Endpoint[]>("list_endpoints"),
  create: (req: CreateEndpointRequest) =>
    request<Endpoint>("create_endpoint", { req }),
  update: (id: number, req: UpdateEndpointRequest) =>
    request<Endpoint>("update_endpoint", { id, req }),
  remove: (id: number) => request<void>("delete_endpoint", { id }),
  reorder: (orderedIds: number[]) =>
    request<void>("reorder_endpoints", { orderedIds }),
  clone: (id: number) => request<Endpoint>("clone_endpoint", { id }),
  test: (id: number, model?: string) =>
    request<EndpointTestResult>("test_endpoint", { id, model }),
  fetchModels: (
    apiUrl: string,
    apiKey: string,
    transformer: string,
    useProxy?: boolean,
  ) =>
    request<string[]>("fetch_endpoint_models", {
      apiUrl,
      apiKey,
      transformer,
      useProxy,
    }),
};
